notify/pagerduty/pagerduty.go

```diff
@@ -143,6 +155,25 @@ func (n *Notifier) encodeMessage(msg *pagerDutyMessage) (bytes.Buffer, error) {
 	return buf, nil
 }
 
+func (n *Notifier) getSecret(ctx context.Context) string {
+	var secret secrets.GenericSecret
+	if !n.conf.ServiceKey.IsZero() {
+		secret = n.conf.ServiceKey
+	} else if !n.conf.RoutingKey.IsZero() {
+		secret = n.conf.RoutingKey
+	}
+	if secret.IsZero() || n.secretsFetcher == nil {
+		return ""
+	}
+
+	sec, err := n.secretsFetcher.FetchSecret(ctx, secret)
+	if err != nil {
+		n.logger.Error("unable to fetch secret", "error", err)
+		return ""
+	}
+	return sec
+}
+
 func (n *Notifier) notifyV1(
```

Comment:
I haven't tested the V1 integration end to end, wondering if we need to consider this in our canaries/tests.